composeApp/src/desktopMain/kotlin/org/ooni/probe/shared/LegacyDirectoryManager.kt

```diff
@@ -0,0 +1,81 @@
+package org.ooni.probe.shared
+
+import co.touchlab.kermit.Logger
+import dev.dirs.ProjectDirectories
+import okio.Path.Companion.toPath
+import java.io.File
+
+class LegacyDirectoryManager(
+    os: DesktopOS,
+) {
+    private val oldProjectDirectories = ProjectDirectories.fromPath("OONI Probe")
+
+    private val legacyPaths = when (os) {
+        DesktopOS.Mac -> listOf(
+            oldProjectDirectories.cacheDir,
+            oldProjectDirectories.dataDir,
+            oldProjectDirectories.configDir,
+            oldProjectDirectories.dataLocalDir,
+            oldProjectDirectories.preferenceDir,
+            System.getProperty("user.home") + "/Library/LaunchAgents/org.ooni.probe-desktop.plist",
+        )
+
+        DesktopOS.Windows -> listOf(
+            oldProjectDirectories.cacheDir
+                .toPath()
+                .parent
+                .toString(),
+            oldProjectDirectories.dataDir
+                .toPath()
+                .parent
+                .toString(),
+        )
+
+        else -> emptyList()
+    }
+
+    fun hasLegacyDirectories(): Boolean =
+        legacyPaths.any { path ->
+            val file = File(path)
+            file.exists() && file.isDirectory && file.listFiles().any()
+        }
+
+    suspend fun cleanupLegacyDirectories(): Boolean {
```

Comment:
I remembered that this should be running on a background CoroutineContext. You should pass the backgroundContext as an argument and wrap the body of this method with `withContext(...) { ... }`.